src/app/comment-period/comment-period.component.ts

```diff
@@ -0,0 +1,46 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CommentPeriod } from '../models/commentperiod';
+import { Subscription } from 'rxjs/Subscription';
+import { PaginationInstance } from 'ngx-pagination';
+
+@Component({
+  selector: 'app-comment-period',
+  templateUrl: './comment-period.component.html',
+  styleUrls: ['./comment-period.component.scss']
+})
+export class CommentPeriodComponent implements OnInit {
+  commentPeriod: CommentPeriod;
+  public loading: boolean;
+  public isDesc: boolean;
+  public column: string;
+  public direction: number;
+
+  public config: PaginationInstance = {
+    id: 'custom',
+    itemsPerPage: 10,
+    currentPage: 1
+  };
+
+  private sub: Subscription;
+
+  constructor( private route: ActivatedRoute, private router: Router) { }
+
+  ngOnInit() {
+    this.loading = true;
+    this.sub = this.route.data.subscribe(
+      (data: { project: CommentPeriod }) => {
```

Comment:
If resolver is updated to store `commentPeriod` in route data, then this should be `data: { commentPeriod: CommentPeriod }`